core/jvmMainModule/src/module-info.java

```diff
@@ -0,0 +1,4 @@
+module kotlinx.collections.immutable {
+    requires java.base;
+    requires transitive kotlin.stdlib;
+}
```

Comment:
New `module-info.java` descriptor. I thought about adding it as `kotlinx.collections.immutable.jvm`, which is the suggested module name in IntelliJ; but I don't think users are using this value yet, because I can't locate it anywhere as an `Automatic-Module-Name`.

Thus, the format here matches other modules, like:
- `kotlinx.coroutines.[core|guava|...]`
- `kotlinx.serialization.[core|json|...]`